scipy/spatial/distance.py

```diff
@@ -1642,6 +1643,76 @@ def sokalsneath(u, v, w=None):
     return float(2.0 * (ntf + nft)) / denom
 
 
+def psi(p, q):
+    """
+    Compute the Population stability (PSI) index (metric) between two
+    two probability arrays.
```

Comment:
If the SciPy community does want this new functionality, and if the appropriate names for `p` and `q` are *probability arrays* rather than generators or point clouds, it does make me wonder about the usual debate between something residing in `stats` vs. `spatial.`

Reply:
Thanks @tylerjereddy for your quick review. I was also wondering what's the best place for this function. The reason I decided to add it to `spatial` was:
- I haven't seen PSI being used in a context of statistical testing (unlike e.g. KS)
- `jensenshannon` is in `spatial` and I'd expect them both to be in the same module.
Although no strong views here and happy to migrate if there is consensus re best place for such functionality.